frontend/pages/queries/edit/EditQueryPage.tsx

```diff
@@ -158,6 +159,24 @@ const EditQueryPage = ({
     }
   );
 
+  /** Pesky bug affecting team level users:
+   - Navigating to queries/:id immediately defaults the user to the first team they're on
+  with the most permissions, in the URL bar because of useTeamIdParam
+  even if the queries/:id entity has a team attached to it
```

Comment:
Hrm, I'll try to take a look at this. I thought that the hook was built to only redirect to the default team if the team id in the URL is not a valid team to which the current user belongs. 

In what you're describing there's no team id param in the URL when the user initially navigates to the page, right? So then the default kicks in but that can result in a mismatch between the default team and the stored query (which we don't know until we get the API response).

Reply:
Exactly that.

The situation arises in which navigating straight to a query/:id on a query whose team_id is set to the user's observer team will still push the user to an admin or maintainer team url param instead. @gillespi314 -- lets pair today?